src/textual/app.py

```diff
@@ -1809,11 +1809,11 @@ async def _close_all(self) -> None:
             if isinstance(screen, Screen) and screen._running:
                 await self._prune_node(screen)
 
-        # Close any remaining nodes
-        # Should be empty by now
+        # Close any remaining nodes. This should be empty by now, unless we couldn't
+        # even compose the app cleanly when starting up.
         remaining_nodes = list(self._registry)
         for child in remaining_nodes:
-            await child._close_messages()
+            await child._close_messages(wait=False)
```

Comment:
You may be misunderstanding this line. This method will always complete processing the message queue. The `wait` argument is so we can wait until it is complete.

Setting `wait=False` feels like it may have consequences.

Reply:
It definitely has consequences: if I remove it, the error gets printed twice again.
I think the point is that I do not wait for the messages to process neither now nor never.

Reply:
> I think you may be on to something. But I think we need to dig a little deeper.

I think the same...
I'll take your thoughts into consideration.

Reply:
Right, but I fear that it will prevent things from shutting down cleanly. It's important to wait for those tasks to finish before exiting.